rgrow/src/models/sdc1d.rs

```diff
@@ -0,0 +1,805 @@
+                ]
+            })
+            .collect::<Vec<PointSafeHere>>();
+
+        points_to_update.sort_unstable();
+        points_to_update.dedup();
+        self.update_points(state, &points_to_update)
+    }
+
+    fn update_monomer_point<S: State + ?Sized>(&self, state: &mut S, scaffold_point: &PointSafe2) {
+        let points = [
+            state.move_sa_w(*scaffold_point),
+            state.move_sa_e(*scaffold_point),
+            PointSafeHere(scaffold_point.0),
+        ]
+        .map(|point| (point, self.event_rate_at_point(state, point)));
+
+        state.update_multiple(&points);
+    }
+
+    /// Fill the energy_bonds array
+    fn fill_energy_array(&mut self) {
+        let num_of_strands = self.strand_names.len();
+
+        // For each *possible* pair of strands, calculate the energy bond
+        for strand_f in 0..(num_of_strands as usize) {
+            let (f_west_glue, f_btm_glue, f_east_glue) = {
+                let glues = self.glues.row(strand_f);
+                (
+                    glues[WEST_GLUE_INDEX],
+                    glues[BOTTOM_GLUE_INDEX],
+                    glues[EAST_GLUE_INDEX],
+                )
+            };
+
+            for strand_s in 0..(num_of_strands as usize) {
+                let (s_west_glue, s_east_glue) = {
+                    let glues = self.glues.row(strand_s);
+                    (glues[WEST_GLUE_INDEX], glues[EAST_GLUE_INDEX])
+                };
+
+                // Calculate the energy between the two strands
+
+                // Case 1: First strands is to the west of second
+                // strand_f    strand_s
+                self.strand_energy_bonds[(strand_f, strand_s)] =
+                    -self.glue_links[(f_east_glue, s_west_glue)];
+
+                // Case 2: First strands is to the east of second
+                // strand_s    strand_f
+                self.strand_energy_bonds[(strand_s, strand_f)] =
+                    -self.glue_links[(f_west_glue, s_east_glue)];
+            }
+
+            // Calculate the binding strength of the starnd with the scaffold
+            self.scaffold_energy_bonds[strand_f] = -self.glue_links[(f_btm_glue, f_btm_glue)];
+        }
+    }
+
+    pub fn monomer_detachment_rate_at_point<S: State + ?Sized>(
+        &self,
+        state: &S,
+        scaffold_point: PointSafe2,
+    ) -> Rate {
+        let strand = state.tile_at_point(scaffold_point);
+
+        // let anchor_tile = self.anchor_tiles[(scaffold_point.0).0]; // FIXME: disabled anchor tiles for now
+
+        // If we are trying to detach the anchor tile
+        // There is no strand, thus nothing to be detached
+        if strand == 0
+        /*|| anchor_tile.0 == scaffold_point */
+        {
+            // FIXME: disabled anchor tiles for now
+            return 0.0;
+        }
+
+        let bond_energy = self.bond_energy_of_strand(state, scaffold_point, strand);
+        self.kf * bond_energy.exp()
```

Comment:
The reason for my previous comments is that this is correct for negative-favored.